content/en/security/application_security/setup/go/dockerfile.md

```
@@ -0,0 +1,258 @@
+---
+title: Building your Go application with Datadog's WAF
+code_lang: docker
+type: multi-code-lang
+code_lang_weight: 10
+further_reading:
+- link: "/security/application_security/how-it-works/"
+  tag: "Documentation"
+  text: "How App and API Protection Works"
+- link: "/security/default_rules/?category=cat-application-security"
+  tag: "Documentation"
+  text: "OOTB App and API Protection Rules"
+- link: "/security/application_security/troubleshooting"
+  tag: "Documentation"
+  text: "Troubleshooting App and API Protection"
+---
+
+# Introduction
+
+App and API Protection for Go installation requirements can be abstract. Moreover the Go toolchain
+cross-compilation and CGO capabilities can make it hard to understand what has to be done precisely.
+
+In these cases, a more precise way to materialize these examples like a Dockerfile can be interesting.
+The goal of this guide is to be a step-by-step guide to a working Dockerfile, tailor-fitted for your usecase.
+
+## Walkthrough
+
+A lot of Dockerfiles found in this guide come from [appsec-go-test-app][4] repository. To try it out, first clone the repository:
+
+```sh
+git clone https://github.com/DataDog/appsec-go-test-app.git
+cd appsec-go-test-app
+```
+
+A list of `Dockerfile` examples can be found in the [`examples/docker`][3] directory.
+Here is an example of it in its simplest form:
+
+```dockerfile
+FROM golang:1 AS build
+WORKDIR /app
+COPY . .
+
+RUN go install github.com/DataDog/orchestrion # Resolved from go.mod dependencies
+
+RUN orchestrion go build -o=main .
+
+FROM debian:bookworm
+COPY --from=build /app/main /usr/local/bin
+
+ENV DD_APPSEC_ENABLED=true
+ENTRYPOINT [ "/usr/local/bin/main" ]
+```
+
+This constitutes the simplest version of a working Dockerfile for a Go application with Datadog's WAF enabled. If this is your first use of [Orchestrion][5]. This dockerfile requires to run `orchestrion pin` beforehand and commit the resulting changes for it to work. Please refer to our [Getting Started for Go][-]
```

```suggestion
This is the simplest version of a working Dockerfile for a Datadog WAF-enabled Go application. If this is your first use of [Orchestrion][5], note that this dockerfile requires you run `orchestrion pin` beforehand and commit the resulting changes. See [Getting Started for Go][-].
```

---

- I think there might have been a mixup in these sentences, so I tried to revise it for clarity.
- I assume there is a link coming for `[-]`.

---

> I assume there is a link coming for [-].
No it was a typo. fixed it

```suggestion
This is the simplest version of a working Dockerfile for a Datadog WAF-enabled Go application. If this is your first use of [Orchestrion][5], note that this dockerfile requires you run `orchestrion pin` beforehand and commit the resulting changes. See [Getting Started for Go][6].
```